torch/__init__.py

```diff
@@ -193,6 +194,29 @@ def _load_global_deps():
 if False:
     import torch._C as _C
 
+# Check to see if we can load C extensions, and if not provide some guidance
+# on what the problem might be.
+try:
+    # _initExtension is chosen (arbitrarily) as a sentinel.
+    from torch._C import _initExtension
+except ImportError:
+    import torch._C as _C_for_compiled_check
```

Comment:
Do you want to report the import error from line 201, or line 203, when _C is not importable at all? 203 will clobber the original error.

Reply:
I think I want the one from 203. (I'm not sure it's possible due to the earlier `from torch._C import *` statements, but for the sake of argument lets say there is a case when it can happen. For now I just mocked it by raising my own `ImportError`) The traceback contains both:

```
Traceback (most recent call last):
  File "/data/users/taylorrobie/repos/pytorch/torch/__init__.py", line 201, in <module>
    from torch._C import _initExtension
ImportError: cannot import name '_initExtension' from 'torch._C' (unknown location)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/data/users/taylorrobie/repos/pytorch/torch/__init__.py", line 204, in <module>
    raise ImportError("Imagine this came from `import torch._C as _C_for_compiled_check`")
ImportError: Imagine this came from `import torch._C as _C_for_compiled_check`
```

Alternatively, if the concern is that the error is complete but confusing we can do a second guard:
```
    try:
        import torch._C as _C_for_compiled_check
    except ImportError:
        raise ImportError("Could not locate the `torch._C` module") from None
```

(Also, TIL about `raise Exception from None` for nested exceptions. Definitely going to add that to the primary exception.)

Reply:
Up to you! I think it is a little confusing, but not worth doing something dramatically different